R/deployApp.R

```diff
@@ -280,7 +280,34 @@ deployApp <- function(appDir = getwd(),
                      application$id), {
       bundlePath <- bundleApp(target$appName, appDir, appFiles,
                               appPrimaryDoc, assetTypeName, contentCategory)
-      bundle <- client$uploadApplication(application$id, bundlePath)
+
+      if (isShinyapps(accountDetails)) {
+
+        # Step 1. Create presigned URL and register pending bundle.
+        bundleSize <- file.info(bundlePath)$size
+
+        # Generate a hex-encoded md5 hash.
+        checkSum <- digest::digest(bundlePath, 'md5', file=TRUE)
+        bundle <- client$createBundle(application$id, "application/x-tar", bundleSize, checkSum)
+
+        # Step 2. Upload Bundle to presigned URL
+        if (!uploadBundle(bundle, bundleSize, bundlePath)) {
+          stop("Could not upload file.")
+        }
+
+        # Step 3. Upload revise bundle status.
+        response <- client$updateBundleStatus(bundle$id, status="ready")
+        if (response$status != 303) {
+          stop("Errors updating bundle status.")
```

Comment:
Is there anything more descriptive we can display here, from e.g. the body of the response? If not let's at least display the HTTP result code.